lib/foreman_maintain/concerns/os_facts.rb

```diff
@@ -0,0 +1,58 @@
+module ForemanMaintain
+  module Concerns
+    module OsFacts
+      OS_RELEASE_FILE = '/etc/os-release'.freeze
+
+      def facts
+        facts = {}
+        tr_map = { '"' => '', "\n" => '' }
+        rexp = Regexp.union(tr_map.keys)
+        file = File.open(OS_RELEASE_FILE)
+        file.readlines.each do |line|
+          key, value = line.gsub(rexp, tr_map).split('=')
```

Comment:
I think lines can have comments which should be skipped.

Reply:
Added one more regex which filters the comments, tested it with adding multiple comments throughout the file and works as expected.